File: src/main/java/wraith/fabricaeexnihilo/api/recipes/ToolRecipe.java
package wraith.fabricaeexnihilo.api.recipes;

import wraith.fabricaeexnihilo.api.crafting.ItemIngredient;
import wraith.fabricaeexnihilo.api.crafting.Loot;

import java.util.ArrayList;
import java.util.List;

public record ToolRecipe(ItemIngredient ingredient, List<Loot> loots) {

    public ToolRecipe(ItemIngredient ingredient) {
        this(ingredient, new ArrayList<>());
    }

    public ToolRecipe(List<Loot> loots) {
        this(ItemIngredient.EMPTY, loots);
    }

    public ToolRecipe() {
        this(ItemIngredient.EMPTY, new ArrayList<>());
    }

}

File: src/main/kotlin/exnihilofabrico/api/recipes/ToolRecipe.kt
package exnihilofabrico.api.recipes

import exnihilofabrico.api.crafting.ItemIngredient
import exnihilofabrico.api.crafting.Lootable

data class ToolRecipe(val ingredient: ItemIngredient = ItemIngredient.EMPTY,
                      val lootables: MutableList<Lootable> = mutableListOf())